lib/rust/prelude/src/range.rs

```diff
@@ -177,5 +178,8 @@ mod test {
         test_merged!([0..=1, 1..=2, 2..=3], [0..=3]);
         test_merged!([0..=2, 3..=5, 1..=4], [0..=5]);
         test_merged!([0..=10, 5..=15], [0..=15]);
+
+        test_merged!([0..2, 0..1], [0..2]);
+        test_merged!([0..10, 4..5, 8..11], [0..11]);
```

Comment:
This is not really documented and tested, but what happens for non-overlapping ranges or multiple partial but not overall overlapping ranges? 
There should be test cases checking the correct output for those. For example, 
* `test_merged!([0..10, 20..30, 40..50], [0..10, 20..30, 40..50]);`
* `test_merged!([0..10, 5..20, 21..30, 25..40], [0..20, 21..30]);`
And I would also add this edge case:
`test_merged!([0..10, 10..20], [0..20]);` (I would assume that to be correct, even if they are only “adjacent” and not overlapping)

Reply:
We are never using it for such cases, but your right, tests should cover it. Adding them there! thanks for catching it!